File: Js_functions/recursion/factorial.js
function factorial (num) {
  if (num === 0) {
    return 1;
  } else {
    return num * factorial(num - 1);
  }
}

if (process.argv.length < 3) {
  console.log('Please input a number to find factorial for!!');
  process.exit(1);
}

try {
  const input = process.argv[2];
  const number = parseInt(input);

  if (isNaN(number)) {
    throw new Error(`"${input}" is not a valid integer input`);
  } else {
    const result = factorial(number);
    console.log(result);
  }
} catch (e) {
  console.log(e + '');
  process.exit(1);
}

File: Js_functions/recursion/factorial.py
import sys

def factorial(num):
    #this function takes in a number and finds the factorial

    if num == 0 :
        return 1
    else:
        return num * factorial(num-1)

if len(sys.argv) < 2:
    print("Please Enter a number")
    sys.exit(1)

try:
    number = int(sys.argv[1])
except ValueError:
    print("Invalid Number")
    sys.exit(1)


result = factorial(number)
print(result)